taf/repository_tool.py

```diff
@@ -264,38 +277,41 @@ def add_targets(self, data, targets_role='targets', files_to_keep=None):
       files_to_keep = []
     # leave all files required by the framework and additional files specified by the user
     files_to_keep.extend(self._required_files)
-
+    # add all repositories defined in repositories.json to files_to_keep
+    files_to_keep.extend(self._get_target_repositories())
     # delete files if they no longer correspond to a target defined
     # in targets metadata and are not specified in files_to_keep
-    for root, _, files in os.walk(self.targets_path):
+    for root, _, files in os.walk(str(self.targets_path)):
       for filename in files:
-        filepath = os.path.join(root, filename)
-        if filepath not in data and filename not in files_to_keep:
-          os.remove(filepath)
+        filepath = Path(root) / filename
+        file_rel_path = str(Path(os.path.relpath(str(filepath), str(self.targets_path))).as_posix())
+        if filepath not in data and file_rel_path not in files_to_keep:
+          filepath.unlink()
 
     targets_obj = self._role_obj(targets_role)
 
     for path, target_data in data.items():
       # if the target's parent directory should not be "targets", create
       # its parent directories if they do not exist
-      target_path = os.path.join(self.targets_path, path)
-      if not os.path.exists(os.path.dirname(target_path)):
-        os.makedirs(os.path.dirname(target_path))
+      target_path = self.targets_path / path
+      target_dir = target_path.parents[0]
+      if not target_dir.exists():
```

Comment:
`target_dir.mkdir(parents=True, exist_ok=True)` should work here

Reply:
Done